packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/launchpad/launchCoin.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  createGenericFile,
+  signerIdentity,
+  createSignerFromKeypair
+} from '@metaplex-foundation/umi'
+import { createUmi } from '@metaplex-foundation/umi-bundle-defaults'
+import { irysUploader } from '@metaplex-foundation/umi-uploader-irys'
+import { DynamicBondingCurveClient } from '@meteora-ag/dynamic-bonding-curve-sdk'
+import { Keypair, PublicKey } from '@solana/web3.js'
+import BN from 'bn.js'
+import { Request, Response } from 'express'
+
+import { config } from '../../config'
+import { logger } from '../../logger'
+import { getConnection } from '../../utils/connections'
+
+interface LaunchCoinRequestBody {
+  name: string
+  symbol: string
+  walletPublicKey: string
+  initialBuyAmount?: number
+  description?: string
+  website?: string
+}
+
+export const launchCoin = async (
+  req: Request<unknown, unknown, LaunchCoinRequestBody> & {
+    file?: Express.Multer.File
+  },
+  res: Response
+) => {
+  try {
+    const { launchpadConfigKey: configKey, solanaFeePayerWallets } = config
+
+    const {
+      name,
+      symbol,
+      description,
+      website,
+      walletPublicKey: walletPublicKeyStr,
+      initialBuyAmount
+    } = req.body
+
+    // file is the image attached via multer middleware (sent from client as a multipart/form-data request)
+    const file = req.file
+    if (!file) {
+      throw new Error('Image file is required.')
+    }
+
+    if (!name || !symbol || !file) {
+      throw new Error(
+        'Invalid metadata arguments. Name, symbol, and image are all required.'
+      )
+    }
+
+    if (!walletPublicKeyStr) {
+      throw new Error(
+        'Invalid wallet public key. Wallet public key is required.'
+      )
+    }
+
+    const walletPublicKey = new PublicKey(walletPublicKeyStr)
+
+    // TODO: get specific addresses with AUDIO in the name
+    const mintKeypair = Keypair.generate()
+
+    const connection = getConnection()
+    const dbcClient = new DynamicBondingCurveClient(connection, 'confirmed')
+
+    // Create Coin Metadata
+    const umi = createUmi(connection.rpcEndpoint).use(irysUploader() as any) // note: something is off with the types with the different umi package versions
+    // Pick a random fee payer to "own" our new coin metadata and pay for the TX
+    const index = Math.floor(Math.random() * solanaFeePayerWallets.length)
+    const feePayer = solanaFeePayerWallets[index]
+    const umiKeypair = umi.eddsa.createKeypairFromSecretKey(feePayer.secretKey)
+    const signer = createSignerFromKeypair(umi, umiKeypair)
+    umi.use(signerIdentity(signer))
+
+    const umiImageFile = createGenericFile(file.buffer, '', {
```

Comment:
I really like the idea of replacing this w/ audius image URLs at some point, but harder to do because our image URLs aren't static. maybe we should add an api.audius.co proxy